MIGRATION-GUIDE.md

```diff
@@ -0,0 +1,84 @@
+
+Notable changes to consider when migrating:
+ - `SplitFactoryProvider` utilizes the React Hooks API, requiring React 16.8.0 or later, while `SplitFactory` is compatible with React 16.3.0 or later.
+
+ - When using the `config` prop with `SplitFactoryProvider`, `factory` and `client` properties in `SplitContext` are `null` in the first render, until the context is updated when some event is emitted on the SDK main client (ready, ready from cache, timeout or update depending on the configuration of the `updateOn<Event>` props of the component). This differs from the previous behavior where `factory` and `client` were immediately available. Nonetheless, it is not recommended to use the `client` and `factory` properties directly as better alternatives are available. For example, use the `useTrack` and `useSplitTreatments` hooks rather than the client's `track` and `getTreatments` methods.
+
+ - Updating the `config` prop in `SplitFactoryProvider` reinitializes the SDK with the new configuration, while `SplitFactory` does not reinitialize the SDK. It is recommended to pass a reference to the configuration object (e.g., via a global variable, `useState`, or `useMemo`) rather than a new instance on each render, to avoid unnecessary reinitializations.
+
+ - Updating the `factory` prop in `SplitFactoryProvider` replaces the current SDK instance, unlike `SplitFactory` where it is ignored.
+
+# React SDK v1.10.0: replace deprecated hooks with new ones
+
+The `useClient`, `useTreatments` and `useManager` hooks have been deprecated since React SDK v1.10.0, and will be removed on a future major release.
+
+We recommend migrating to the new versions `useSplitClient`, `useSplitTreatments` and `useSplitManager` respectively, which provide a more flexible API:
+
+- They accept extra optional parameters, `updateOnSdkReady`, `updateOnSdkReadyFromCache`, `updateOnSdkTimedout` and `updateOnSdkUpdate`, which enable to control when the hook updates the component. For example, you can set `updateOnSdkReady` to `true`, which is `false` by default, to update the component when an `SDK_UPDATE` event is emitted. This is useful when you want to avoid unnecessary re-renders of your components.
+
+- They return an object containing the SDK status properties. These properties are described in the ['Subscribe to events and changes' section](https://help.split.io/hc/en-us/articles/360038825091-React-SDK#subscribe-to-events-and-changes) and enable conditional rendering of components based on the SDK status, eliminating the need to access the Split context or using the client's `ready` promise or event listeners. For example, you can show a loading spinner while the SDK is not ready, and use the `treatments` result to render the variants of your App once the SDK is ready.
+
```

Comment:
```suggestion

The new hooks accept the same parameters as their depreciated counterparts, but return an object containing the SDK status properties, as shown below:

```

Reply:
I am not sure how to include this new paragraph without looking redundant with lines 22 and 24. I will chat with you later.

Reply:
> I am not sure how to include this new paragraph without looking redundant with lines 22 and 24. I will chat with you later.

Ah, I get it. On first read, it just looked like the code snippet was not introduced... How about something like this:
```suggestion
The usage of the new hooks is shown below:
```
Alternately, we could say, "The initialization of the new hooks is shown below" or something more appropriate. The point is just to briefly introduce the code snippet.

Reply:
Yes. I like "The usage of the new hooks is shown below:". I will include it